source/common/http/status.h

```diff
@@ -109,5 +109,37 @@ ABSL_MUST_USE_RESULT bool isCodecClientError(const Status& status);
  */
 Http::Code getPrematureResponseHttpCode(const Status& status);
 
+/**
+ * Macro that checks return value of expression that results in Status and returns from
+ * the current function is status is not OK.
+ *
+ * Example usage:
+ *   Status foo() {
+ *     RETURN_IF_ERROR(bar());
+ *     return okStatus();
+ *   }
+ */
+
+#define RETURN_IF_ERROR(expr)                                                                      \
+  if (::Envoy::Http::Details::StatusAdapter adapter{(expr)}) {                                     \
```

Comment:
wrap with `do {} while(0)` so it won't allow missing semicolon at the end.

Reply:
ok